### PR TITLE
removes duplicated code for http provider

### LINKED ISSUE #516: remove follow http client configuration from service library
Please create the related github issue and link it here.

_Originally posted by @zengin in https://github.com/microsoftgraph/msgraph-sdk-java/pull/501#r498978484_

### LINKED ISSUE #517: remove authentication provider from service library as already done by middleware
Same here: link to corresponding github issue.

To save time, I like @MIchaelMainer's tactic of adding the title (with an optional quick summary) for creating the link as a reference quickly. The details can be filled afterwards.

_Originally posted by @zengin in https://github.com/microsoftgraph/msgraph-sdk-java/pull/501#discussion_r498981936_

[AB#5761](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/5761)

### PR DESCRIPTION
**Pull request by baywet**
Supersedes #202 
fixes #516 
fixes #517 

### Changes proposed in this pull request
-

### Other links

-

### CODE REVIEW

**Comment by baywet** [comment]
TODO: move the files excludes in the azure pipeline definition to a higher level so we don't have to exclude them anymore

**Comment by MIchaelMainer** [comment]
I'm adding a feature request to GH to **turn off notifications while a PR is in draft state**. So I setup custom notification setting to see if it will turn off commit emails until you assign me as reviewer:

image

I think just by posting this it is considered participation and I will continue to get emails so I will reset. I'll let you.

**Review by MIchaelMainer — APPROVED** [review]

File: src/main/java/com/microsoft/graph/http/CoreHttpProvider.java [hunk 1]
```diff
@@ -102,41 +101,47 @@
...
+			this.serializer = serializer;
+			this.executors = executors;
+			this.logger = logger;
+			this.corehttpClient = httpClient;
+		}
 	}
 
 	/**
-	 * Creates the CoreHttpProvider
+	 * Creates the DefaultHttpProvider
```
[the remark this answers is not in the document]
Comment (baywet) [reply src/main/java/com/microsoft/graph/http/CoreHttpProvider.java:127]: not at this time, I think it's a wrong string replace. thanks for catching that!

[the remark this answers is not in the document]
Comment (baywet) [reply src/main/java/com/microsoft/graph/http/CoreHttpProvider.java:127]:
```suggestion
	 * Creates the CoreHttpProvider
```

**Pull request merged**